server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java

```diff
@@ -0,0 +1,126 @@
+package com.stnetix.ariaddna.servercore.tools;
+
+
+import com.stnetix.ariaddna.commonutils.exception.AriaddnaException;
+import com.stnetix.ariaddna.commonutils.logger.AriaddnaLogger;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+/**
+ * Created by alexkotov on 27.02.17.
+ */
+public class ServiceTree {
+    private HashSet<ServiceNode> nodes;
+    private ArrayList<ServiceNode> startQueue;
+
+    private static AriaddnaLogger logger = AriaddnaLogger.getLogger(ServiceTree.class);
+
+    public ServiceTree() {
+        nodes = new HashSet<>();
+    }
+
+    public void addService(IService service, IService parent) throws ServiceNotFoundException {
```

Comment:
You did not understand me. In current code you use serial initialization. It is true if own service have only one parent, for example: server-core depends on common-utils.
But we will have services with many dependencies, for example: VUFS Synchronization Service depends on Balancer Service and Allocation Storage Service, but Balancer Service and Allocation Storage Service is independent services.